### PR TITLE
fix: correctly use merlin's pipeline

### PR DESCRIPTION
**Pull request by rgrinberg**
do not reuse pipelines unless we are using the same pipeline multiple
times in a row

@ulugbekna let's get a quick fix in and work on optimizing in #901 parallel. This bug is likely very annoying to users.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 1.14.2
+
+## Fixes
+
+- Fix random requests failing after switching documents (#901, fixes #898)
```
Comment (ulugbekna) [inline CHANGES.md:5]:
```suggestion
- Fix random requests failing after switching documents (#904, fixes #898)
```

**Review by ulugbekna — CHANGES_REQUESTED** [review]
Looks correct. Indeed fixes the go-to-definition for me.

Noteworthy: there's `Mpipeline.make` used in `action_type_annotate.ml`. Could this trip up `Single_pipeline`? I couldn't reproduce the initial bug anyway.

**Comment by rgrinberg** [comment]
>  there's Mpipeline.make used in action_type_annotate.ml. Could this trip up Single_pipeline? I couldn't reproduce the initial bug anyway.

It should be fine b/c it creates the pipeline inside the merlin thread.

**Comment by ulugbekna** [comment]
Shall we merge this?

**Comment by rgrinberg** [comment]
Let me try and fix the tests.

Rudi.

**Pull request merged**